### PR TITLE
TOML: Implement IconProvider

### PR DESCRIPTION
**Pull request by Turbo87**
We should only show the Cargo logo for files that are actually called "Cargo.toml", other TOML files will keep the default text file icon for now.

### CODE REVIEW

**Comment by pavel-v-chernykh** [comment]
Oops, I've introduced some conflicts by my changes. Sorry about that.

**Comment by Turbo87** [comment]
not a problem, I'll rebase this branch

**Comment by matklad** [comment]
@Turbo87 what is the reason behind renaming of the icons files (`cargo32.png -> cargo@2.png`)?

The former variant is consistent with `rust16.png`, `rust32.png` and looks more self documenting.

File: src/org/toml/lang/icons/TomlIconProvider.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+public class TomlIconProvider : IconProvider() {
+    override fun getIcon(element: PsiElement, flags: Int): Icon? {
+        return when(element) {
+            is TomlFile -> getFileIcon(element)
+            else -> null
+        }
+    }
+
+    fun getFileIcon(element: TomlFile): Icon? {
```
Comment (matklad) [inline src/org/toml/lang/icons/TomlIconProvider.kt]: In `Kotlin` everything is public by default. Is this `fun` meant to be private? If it is, then use `private fun ...`

File: src/org/toml/lang/icons/TomlIcons.kt [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
 
+import com.intellij.icons.AllIcons
 import com.intellij.openapi.util.IconLoader
 import javax.swing.Icon
 
 
 public object TomlIcons {
-    public val NORMAL: Icon = IconLoader.getIcon("/org/toml/icons/cargo16.png")
-    public val BIG: Icon = IconLoader.getIcon("/org/toml/icons/cargo32.png")
+    public val FILE = AllIcons.FileTypes.Text;
```
Comment (matklad) [inline src/org/toml/lang/icons/TomlIcons.kt]: I'll check if IDEA has an inspection for semicolons =)

File: src/org/toml/lang/icons/TomlIconProvider.kt (hunk 1 above)
Comment (Turbo87) [inline src/org/toml/lang/icons/TomlIconProvider.kt]: There is currently no one calling the method externally, but I don't see why that shouldn't happen if you already have a `TomlFile` instance and don't want to perform the unnecessary pattern matching in the `getIcon()` method

File: src/org/toml/lang/icons/TomlIcons.kt (hunk 2 above)
Comment (Turbo87) [inline src/org/toml/lang/icons/TomlIcons.kt]: :smile:

**Comment by Turbo87** [comment]
@matklad the `@2x` suffix is usually used on Apple platforms for retina support and [intellij-community](https://github.com/JetBrains/intellij-community/tree/master/platform/icons/src/fileTypes) seems to use that convention too. I guess the rust icons should follow that convention too. Unfortunately I don't have a retina screen so I can't test this.

File: src/org/toml/lang/icons/TomlIcons.kt (hunk 2 above)
Comment (matklad) [inline src/org/toml/lang/icons/TomlIcons.kt]: Nope, it has not at the moment :(

The only thing you can do is to set semicolon color to bold blood red :)

**Comment by pavel-v-chernykh** [comment]
Yep, It makes sense to rename icons to ```rust@2x.png```. I've renamed rust16.png to rust.png and rust32.png to rust@2x.png and icons looks sharper on retina.

**Comment by pavel-v-chernykh** [comment]
Take a look 

* ```rust16.png``` ```rust32.png```

without_suffix

* ```rust@2x.png```

with_2x_suffix

**Comment by matklad** [comment]
@Turbo87 @pavel-v-chernykh thanks for clarification!

**Pull request merged**